sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -136,19 +185,45 @@ export class MSFAANumberSharedService {
           updatedAt: now,
         },
       );
-      // Create the new MSFAA record for the student.
-      const newMSFAANumber = new MSFAANumber();
-      newMSFAANumber.msfaaNumber = await this.consumeNextSequence(
-        offeringIntensity,
-      );
-      newMSFAANumber.student = { id: studentId } as Student;
-      newMSFAANumber.referenceApplication = {
-        id: referenceApplicationId,
-      } as Application;
-      newMSFAANumber.offeringIntensity = offeringIntensity;
-      newMSFAANumber.creator = { id: auditUserId } as User;
-      newMSFAANumber.createdAt = now;
-      await entityManager.getRepository(MSFAANumber).save(newMSFAANumber);
+      // The reactivated msfaaNumber.
+      let newActiveMSFAA = {
+        id: options?.existingMSFAA.id,
+      } as MSFAANumber;
+      // If there is not a MSFAA number to be re-activated.
+      if (!options?.existingMSFAA) {
+        // Create the new MSFAA record for the student.
+        newActiveMSFAA = new MSFAANumber();
+        newActiveMSFAA.msfaaNumber = await this.consumeNextSequence(
+          offeringIntensity,
+        );
+        newActiveMSFAA.student = { id: studentId } as Student;
+        newActiveMSFAA.referenceApplication = {
+          id: referenceApplicationId,
+        } as Application;
+        newActiveMSFAA.offeringIntensity = offeringIntensity;
+        newActiveMSFAA.creator = { id: auditUserId } as User;
+        newActiveMSFAA.createdAt = now;
+        await entityManager.getRepository(MSFAANumber).save(newActiveMSFAA);
+      } else {
+        // Reactivate this msfaa record.
+        const systemUser = await this.systemUsersService.systemUser();
+        const updateResult = await this.dataSource
+          .getRepository(MSFAANumber)
+          .update(options.existingMSFAA.id, {
```

Comment:
maybe u can use `newActiveMSFAA` here

Reply:
@ann-aot shouldn't I use `options.existingMSFAA` instead of `newActiveMSFAA` to keep it consistent with other parts of the query?

Reply:
Either way would work for me, @sh16011993 please pick one and move on.

Reply:
@ann-aot Using `newActiveMSFAA` or `options.existingMSFAA` instead of `options.existingMSFAA.id` won't work because I need to update based only on the msfaa id and not other properties which are a part of `newActiveMSFAA`.

Reply:
my comment was to use `newActiveMSFAA .id`, but was a suggestion.  resolving